internal/host/statemachine.go

```diff
@@ -257,16 +268,78 @@ func NewHostStateMachine(th *transitionHandler) stateswitch.StateMachine {
 
 	// Prepare for installation
 	sm.AddTransition(stateswitch.TransitionRule{
-		TransitionType: TransitionTypePrepareForInstallation,
-		Condition:      th.IsValidRoleForInstallation,
+		TransitionType: TransitionTypeRefresh,
+		Condition:      stateswitch.And(If(ValidRoleForInstallation), If(IsConnected), If(ClusterPreparingForInstallation)),
 		SourceStates: []stateswitch.State{
 			stateswitch.State(models.HostStatusKnown),
 		},
 		DestinationState: stateswitch.State(models.HostStatusPreparingForInstallation),
-		PostTransition:   th.PostPrepareForInstallation,
+		PostTransition:   th.PostRefreshHost(statusInfoPreparingForInstallation),
+	})
+
+	// Install host
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusPreparingSuccessful),
+		},
+		Condition:        stateswitch.And(If(IsConnected), If(ClusterInstalling)),
+		DestinationState: stateswitch.State(models.HostStatusInstalling),
+		PostTransition:   th.PostRefreshHost(statusInfoInstalling),
 	})
 
 	// Refresh host
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusPreparingForInstallation),
+		},
+		Condition:        stateswitch.And(If(IsConnected), If(InstallationDiskSpeedCheckSuccessful), If(ClusterPreparingForInstallation)),
+		DestinationState: stateswitch.State(models.HostStatusPreparingSuccessful),
+		PostTransition:   th.PostRefreshHost(statusInfoHostPreparationSuccessful),
+	})
+
+	installationDiskSpeedCheckFailed := stateswitch.Not(If(SufficientOrUnknownInstallationDiskSpeed))
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusPreparingForInstallation),
+		},
+		Condition:        stateswitch.And(If(IsConnected), installationDiskSpeedCheckFailed),
+		DestinationState: stateswitch.State(models.HostStatusInsufficient),
+		PostTransition:   th.PostRefreshHost(statusInfoNotReadyForInstall),
+	})
+
+	sm.AddTransition(stateswitch.TransitionRule{
+		TransitionType: TransitionTypeRefresh,
+		SourceStates: []stateswitch.State{
+			stateswitch.State(models.HostStatusPreparingForInstallation),
+		},
+		Condition:        stateswitch.And(If(IsConnected), If(SufficientOrUnknownInstallationDiskSpeed), stateswitch.Not(If(ClusterPreparingForInstallation))),
```

Comment:
Host that is stuck on fio needs to move to Insufficient in case cluster has been moved to Ready due to stage timeout

Reply:
OK.  Probably as part of the command that will fail on timeout